### ISSUE TITLE
Prev and Next month date Hidden in Current Month in JTAppleCalendar

### ISSUE THREAD

**Issue by Harshad290194** [patchthecode/JTAppleCalendar#522]
I have an issue in show prev and next date in Current month calendar. how i can disable that prev and next date of next and pre month date to be display in current date. How i can Disable that date in rad box... please give me a solution for my issue.
![img_2549 png](https://user-images.githubusercontent.com/17276435/28966464-f813061c-7933-11e7-8642-b535f203bccc.png)

**Comment by patchthecode** [comment]
use the calendarView `shouldSelect` delegate function

```swift
func calendar(_ calendar: JTAppleCalendarView, shouldSelectDate date: Date, cell: JTAppleCell?, cellState: CellState) -> Bool { 
   if cellState.dateBelongsTo != thisMonth {
        return false
   }
   return true 
}
```

closing issue. let me know if youre still stuck

**Comment by joseph-francis** [comment]
That doesn't work @patchthecode

**Comment by patchthecode** [comment]
@joseph-francis 

1. What are you trying to do?
2. What code have you done to try and do it?

**Comment by joseph-francis** [comment]
@patchthecode I wanted to not show the dates of previous and after months. I'm just hiding those cells in cellForitemAt. It's working now.

I just wanted to say that this library is great. Thank you and god bless you brother!

Also, I think there's a minor problem. What I have is a nib file with JTAppleCalendarView (the collectionView). Let's call this "parentView". The CalendarCell (subclass of JTAppleCell, the cell used for the collectionView) is also in a nib file. Let's call this "cellView" The owner of "parentView" is a view controller. The problem is that the "cellView" registration slows down the loading of the "parentView". I benchmarked this with a regular UICollectionView with cells - this loads the view faster when compared to JTAppleCalendarView. I also noticed that if I remove the JTAppleCalendarView from "parentView", the loading speeds up. If you like, I'll open a new issue with this question. Please let me know if you were able to duplicate this.

Thank you so much!

**Comment by patchthecode** [comment]
@joseph-francis sounds good. is it possible you can upload on your github a sample project showing this?
i'm always up for making things more efficient.

**Comment by joseph-francis** [comment]
https://github.com/joseph-francis/Example
Now, I suspect if it's the top NSLayoutConstraint animation is causing the slowness, or adding the subview.

**Comment by patchthecode** [comment]
@joseph-francis i checked out your code.

the speeds are as follows:
```
FastView time taken: 0.00717306137084961
calendarView time taken: 0.00909900665283203
calendarView time taken: 0.00309598445892334
FastView time taken: 0.000952005386352539
calendarView time taken: 0.00271701812744141
calendarView time taken: 0.0024569034576416
```

Just wondering, since those times are very small, is it impacting your app? Or are you just looking wondering why the time difference?

**Comment by joseph-francis** [comment]
Well, that's odd. That's not the time I got when I ran. I got like 0.03 (fastView) vs 0.6 (calendarView). I'm running this on my iphone. Are you running in a simulator? Did you change anything? If so, it could really help.

**Comment by patchthecode** [comment]
@joseph-francis ive tried it on simulator. Let me try it on iPhone and check it out. brb.

**Comment by patchthecode** [comment]
new results from iPhone SE:
```
calendarView time taken: 0.0131890773773193
FastView time taken: 0.00334501266479492
FastView time taken: 0.00117099285125732
FastView time taken: 0.00266897678375244
FastView time taken: 0.00257289409637451
FastView time taken: 0.0025789737701416
FastView time taken: 0.00260496139526367
calendarView time taken: 0.00994002819061279
calendarView time taken: 0.010003924369812
calendarView time taken: 0.0100749731063843
calendarView time taken: 0.0105949640274048
```

so a couple questions:

1. The sample collection view you are adding just has `1 section` with `3 items` in it. In the calendar view however, there are a lot of calculations going-on on setup in the collectionViewLayout file.
If think this accounts for the difference?
2. Just wondering, is this speed hindering your app? (it looks too small to matter i think?)

**Comment by joseph-francis** [comment]
Hey, I was able to pinpoint what's happening, but don't know how to solve. On my actual project, if I comment out `collectionView.calendarDataSource = self` and `collectionView.calendarDelegate = self`, it works faster. I'm setting the delegate and datasource in the awakeFromNib method. After I add the parentView (from the Example), I animate the top constraint. The animation is only over after the delegate, datasource and the cells are loaded. This is what causing the delay. I'm not being able to replicate this problem in the project I gave you, but I'm sure that this is causing the problem. Where else do I set the datasource and delegate? I have tried setting the dataSource and delegate after the animation complete and reloading data, but that's not an elegant solution.
